clojure-mode.el

```diff
@@ -327,7 +327,44 @@ If JUSTIFY is non-nil, justify as well as fill the paragraph."
         (do-auto-fill)))))
 
 
+;;; #_ comments font-locking
+;; Code heavily borrowed from Slime.
+;; https://github.com/slime/slime/blob/master/contrib/slime-fontifying-fu.el#L186
+(defconst clojure--comment-macro-regexp
+  (rx (or (and "#_" (* " ") (group-n 1 (not (any " "))))
+          (group-n 1 "(comment" symbol-end)))
+  "Regexp matching the start of a comment sexp.
+The beginning of match-group 1 should be before the sexp to be marked as a
+comment.  The end of sexp is found with `clojure-forward-logical-sexp'.")
+
+(defun clojure--search-comment-macro-internal (limit)
+  (when (search-forward-regexp clojure--comment-macro-regexp limit t)
+    (let* ((md (match-data))
+           (start (match-beginning 1))
+           (state (syntax-ppss start)))
+      ;; inside string or comment?
+      (if (or (nth 3 state)
+              (nth 4 state))
+          (clojure--search-comment-macro-internal limit)
+        (goto-char start)
+        (clojure-forward-logical-sexp 1)
+        ;; Data for (match-end 1).
+        (setf (elt md 3) (point))
```

Comment:
I almost always use it to destructively change a sequence or struct. :-)
It's considerably more convenient than something like `(setcar (cdr (cdr (cdr md))) start)`.